cocos/2d/utils/dynamic-atlas/atlas-manager.ts

```diff
@@ -179,7 +179,7 @@ export class DynamicAtlasManager extends System {
         texture: DynamicAtlasTexture;
     } | null {
         if (EDITOR_NOT_IN_PREVIEW) return null;
-        if (!this._enabled || this._atlasIndex === this._maxAtlasCount
+        if (!this._enabled || this._atlasIndex >= this._maxAtlasCount
```

Comment:
When can this._atlasIndex greater than this._maxAtlasCount?

Reply:
This might be safer